jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/BhandariKDisjointShortestPaths.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * (C) Copyright 2018-2018, by Assaf Mizrachi and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.shortestpath;
+
+import java.util.*;
+import java.util.stream.*;
+
+import org.jgrapht.*;
+import org.jgrapht.alg.interfaces.*;
+import org.jgrapht.graph.*;
+
+/**
+ * An implementation of Bhandari algorithm for finding $K$ edge-<em>disjoint</em> shortest paths.
+ * The algorithm determines the $k$ disjoint shortest simple paths in increasing order of
+ * weight. Weights can be negative (but no negative cycle is allowed). Only directed simple graphs
+ * are allowed.
+ *
+ * <p>
+ * The algorithm is running $k$ sequential Bellman-Ford iterations to find the shortest path at each step.
+ * Hence, yielding a complexity of $k$*O(Bellman-Ford).
+ * 
+ * <p>
+ * For further reference see <a href="https://www.nas.ewi.tudelft.nl/people/Fernando/papers/Wiley.pdf">

```

Comment:
The way this is written, nobody will know what the 'main reference of your class' was when the URL stops working...